drivers/storage/portworx/manifest/remote_test.go

```diff
@@ -0,0 +1,216 @@
+package manifest
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	"github.com/stretchr/testify/require"
+	"k8s.io/api/core/v1"
+)
+
+func TestRemoteManifestWithMatchingVersion(t *testing.T) {
+	pxVersion := "3.2.1"
+	expectedManifestURL := manifestURLFromVersion(pxVersion)
+	httpGet = func(url string) (*http.Response, error) {
+		require.Equal(t, expectedManifestURL, url)
+		return &http.Response{
+			Body: ioutil.NopCloser(bytes.NewReader([]byte(`
+version: 3.2.1
+components:
+  stork: stork/image:3.2.1
+`))),
+		}, nil
+	}
+
+	cluster := &corev1alpha1.StorageCluster{
+		Spec: corev1alpha1.StorageClusterSpec{
+			Image: "px/image:" + pxVersion,
+		},
+	}
+
+	r, err := newRemoteManifest(cluster).Get()
+	require.NoError(t, err)
+	require.Equal(t, pxVersion, r.PortworxVersion)
+	require.Equal(t, "stork/image:3.2.1", r.Components.Stork)
+}
+
+func TestRemoteManifestWithoutMatchingVersion(t *testing.T) {
+	pxVersion := "3.2.1"
+	expectedManifestURL := manifestURLFromVersion(pxVersion)
+	httpGet = func(url string) (*http.Response, error) {
+		require.Equal(t, expectedManifestURL, url)
+		return &http.Response{
+			Body: ioutil.NopCloser(bytes.NewReader([]byte(`
+version: 3.2.1.1
+components:
+  stork: stork/image:3.2.1.1
+`))),
+		}, nil
+	}
+
+	cluster := &corev1alpha1.StorageCluster{
+		Spec: corev1alpha1.StorageClusterSpec{
+			Image: "px/image:" + pxVersion,
+		},
+	}
+
+	// Even if the image version does not match the one
+	// returned by manifest, we still return those versions
+	r, err := newRemoteManifest(cluster).Get()
```

Comment:
Ideally this should not happen if we update the installer correctly. 
Just in case it happens we take whatever the installer returns ignoring the px version. As we already have the px version from image, we will use that but use the components returned by the installer.